manager/dispatcher/dispatcher.go

```diff
@@ -342,6 +342,19 @@ func (d *Dispatcher) Stop() error {
 	d.cancel()
 	d.mu.Unlock()
 
+	d.processUpdatesLock.Lock()
+	// when we called d.cancel(), there may have been waiters currently
+	// waiting. they would be forever sleeping if we didn't call Broadcast here
+	// to wake them. However, once we've canceled the context (d.cancel()), no
+	// more waits will start after this Broadcast, because before waiting they
+	// check if the context is canceled.
+	//
+	// if that context cancelation check were not present, it would be possible
```

Comment:
Its not clear from this description that the waiters actually go on wait() holding the rpcRW. Do we want to clarify that?